src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/SecurityServiceDispatcher.cs

```diff
@@ -492,6 +492,176 @@ public Task OpenAsync(CancellationToken token)
         }
     }
 
+    internal abstract class SecurityDuplexChannel<UChannel> : IServiceChannelDispatcher where UChannel : class
+    {
+        private readonly IDuplexChannel innerDuplexChannel;
+        private readonly IServiceProvider _serviceProvider;
+        public SecurityDuplexChannel(SecurityServiceDispatcher serviceDispatcher, IDuplexChannel innerChannel, SecurityProtocol securityProtocol, SecurityListenerSettingsLifetimeManager settingsLifetimeManager)
+          //  : base(channelManager, innerChannel, securityProtocol, settingsLifetimeManager)
+        {
+           innerDuplexChannel = innerChannel;
+            SecurityProtocol = securityProtocol;
+           _serviceProvider = InnerDuplexChannel.GetProperty<IServiceScopeFactory>().CreateScope().ServiceProvider;
+        }
+
+        public EndpointAddress RemoteAddress
+        {
+            get { return this.innerDuplexChannel.RemoteAddress; }
+        }
+
+        public Uri Via
+        {
+            get { return this.innerDuplexChannel.Via; }
+        }
+
+        protected IDuplexChannel InnerDuplexChannel
+        {
+            get { return this.innerDuplexChannel; }
+        }
+
+        internal SecurityProtocol SecurityProtocol { get; set; }
+
+      //  public IReplyChannel OuterChannel { get; private set; }
+
+        public T GetProperty<T>() where T : class
+        {
+            T tObj = _serviceProvider.GetService<T>();
+            if (tObj == null)
+                return InnerDuplexChannel.GetProperty<T>();
```

Comment:
Can be simplifed to `return tObj ?? InnerDuplexChannel.GetProperty<T>();`

Reply:
done